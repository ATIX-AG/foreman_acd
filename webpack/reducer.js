import { combineReducers } from 'redux';

import parameterSelectionParameters from './components/ParameterSelection/ParameterSelectionReducer';
import applicationDefinitionConf from './components/ApplicationDefinition/ApplicationDefinitionReducer';
import applicationDefinitionImportConf from './components/ApplicationDefinitionImport/ApplicationDefinitionImportReducer';
import applicationInstanceConf from './components/ApplicationInstance/ApplicationInstanceReducer';
import applicationInstanceReport from './components/ApplicationInstanceReport/ApplicationInstanceReportReducer';
import syncGitRepoConf from './components/SyncGitRepo/SyncGitRepoReducer';

import {
  APPLICATION_DEFINITION_FOREMAN_PARAMETER_SELECTION_MODAL_CLOSE,
  APPLICATION_DEFINITION_ANSIBLE_PARAMETER_SELECTION_MODAL_CLOSE,
} from './components/ApplicationDefinition/ApplicationDefinitionConstants';

import {
  APPLICATION_INSTANCE_FOREMAN_PARAMETER_SELECTION_MODAL_CLOSE,
  APPLICATION_INSTANCE_ANSIBLE_PARAMETER_SELECTION_MODAL_CLOSE,
} from './components/ApplicationInstance/ApplicationInstanceConstants';

const rootReducer = (state = {}, action) => {

  const param_state = parameterSelectionParameters(state.parameterSelectionParameters, action);
  const app_ins_report_state = applicationInstanceReport(state.applicationInstanceReport, action);

  if (action.type == APPLICATION_DEFINITION_FOREMAN_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.parameterSelection = param_state.parameters;
  }
  if (action.type == APPLICATION_DEFINITION_ANSIBLE_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.parameterSelection = param_state.parameters;
  }
  const app_def_state = applicationDefinitionConf(state.applicationDefinitionConf, action);
  const app_def_import_state = applicationDefinitionImportConf(state.applicationDefinitionImportConf, action);

  if (action.type == APPLICATION_INSTANCE_FOREMAN_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.parameterSelection = param_state.parameters;
  }
  if (action.type == APPLICATION_INSTANCE_ANSIBLE_PARAMETER_SELECTION_MODAL_CLOSE) {
    action.payload.parameterSelection = param_state.parameters;
  }
  const app_ins_state = applicationInstanceConf(state.applicationInstanceConf, action);

  const sync_git_repo_state = syncGitRepoConf(state.syncGitRepoConf, action);

  return {
    applicationDefinitionConf: app_def_state,
    applicationDefinitionImportConf: app_def_import_state,
    applicationInstanceConf: app_ins_state,
    parameterSelectionParameters: param_state,
    applicationInstanceReport: app_ins_report_state,
    syncGitRepoConf: sync_git_repo_state,
  };
};

export default rootReducer;
