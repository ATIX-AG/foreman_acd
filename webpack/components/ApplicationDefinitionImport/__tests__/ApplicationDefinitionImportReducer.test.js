import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import reducer, { initialState } from '../ApplicationDefinitionImportReducer';

import {
  editState,
  initApplicationDefinitionImportPayload,
  changeEditServicePayload,
} from '../__fixtures__/applicationDefinitionImportReducer.fixtures';

import {
  APPLICATION_DEFINITION_IMPORT_INIT,
  APPLICATION_DEFINITION_IMPORT_SERVICE_EDIT_CHANGE,
} from '../ApplicationDefinitionImportConstants';

const fixtures = {
  'should return initial state': {
    state: initialState,
    action: {
      type: undefined,
      payload: {},
    },
  },

  'should initialize component': {
    state: initialState,
    action: {
      type: APPLICATION_DEFINITION_IMPORT_INIT,
      payload: initApplicationDefinitionImportPayload,
    },
  },
  'should change edit service': {
    state: editState,
    action: {
      type: APPLICATION_DEFINITION_IMPORT_SERVICE_EDIT_CHANGE,
      payload: changeEditServicePayload,
    },
  },
};

describe('ApplicationDefinitionImportReducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
