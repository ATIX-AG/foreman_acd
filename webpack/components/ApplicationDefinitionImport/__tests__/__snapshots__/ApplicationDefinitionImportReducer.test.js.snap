// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationDefinitionImportReducer should change edit service 1`] = `
Object {
  "ansiblePlaybookServices": Array [
    Object {
      "ansibleGroup": "webservers",
      "ansibleParameters": Array [
        Object {
          "id": 0,
          "name": "dummy_var",
          "value": "0",
        },
      ],
      "description": "",
      "foremanParameters": Array [
        Object {
          "description": "",
          "id": 1,
          "locked": false,
          "name": "CP",
          "type": "computeprofile",
          "value": "1",
        },
        Object {
          "description": "",
          "id": 2,
          "locked": true,
          "name": "LE",
          "type": "lifecycleenv",
          "value": "1",
        },
      ],
      "hostgroup": "1",
      "id": 1,
      "maxCount": "",
      "minCount": "2",
      "name": "web",
    },
    Object {
      "ansibleGroup": "dbservers",
      "ansibleParameters": Array [
        Object {
          "id": 0,
          "name": "mysqlservice",
          "value": "mysqld",
        },
        Object {
          "id": 1,
          "locked": true,
          "name": "mysql_port",
          "value": "3306",
        },
        Object {
          "id": 2,
          "name": "dbuser",
          "value": "webapp",
        },
        Object {
          "id": 3,
          "name": "dbname",
          "value": "ANSAP01",
        },
        Object {
          "id": 4,
          "name": "upassword",
          "value": "Bond@007",
        },
        Object {
          "id": 5,
          "name": "masterpassword",
          "value": "MySQL@007",
        },
      ],
      "backup": Object {
        "ansibleGroup": "dbservers",
        "ansibleParameters": Array [
          Object {
            "id": 0,
            "name": "mysqlservice",
            "value": "mysqld",
          },
          Object {
            "id": 1,
            "locked": true,
            "name": "mysql_port",
            "value": "3306",
          },
          Object {
            "id": 2,
            "name": "dbuser",
            "value": "webapp",
          },
          Object {
            "id": 3,
            "name": "dbname",
            "value": "ANSAP01",
          },
          Object {
            "id": 4,
            "name": "upassword",
            "value": "Bond@007",
          },
          Object {
            "id": 5,
            "name": "masterpassword",
            "value": "MySQL@007",
          },
        ],
        "description": "",
        "foremanParameters": Array [],
        "hostgroup": "1",
        "id": 2,
        "maxCount": "",
        "minCount": "1",
        "name": "db",
      },
      "description": "",
      "foremanParameters": Array [],
      "hostgroup": "1",
      "id": 2,
      "maxCount": "",
      "minCount": "1",
      "name": "helloworld",
    },
  ],
  "columns": Array [
    Object {
      "cell": Object {
        "formatters": Array [
          null,
        ],
      },
      "header": Object {
        "formatters": Array [
          null,
        ],
        "label": "Service name",
        "props": Object {
          "index": 3,
          "style": Object {
            "width": "20%",
          },
        },
      },
      "property": "name",
    },
    Object {
      "cell": Object {
        "formatters": Array [
          null,
        ],
      },
      "header": Object {
        "formatters": Array [
          null,
        ],
        "label": "Hostgroup",
        "props": Object {
          "index": 2,
          "style": Object {
            "width": "20%",
          },
        },
      },
      "property": "hostgroup",
    },
  ],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "name": false,
}
`;

exports[`ApplicationDefinitionImportReducer should initialize component 1`] = `
Object {
  "ansiblePlaybookServices": Array [
    Object {
      "ansibleGroup": "webservers",
      "ansibleParameters": Array [
        Object {
          "id": 0,
          "name": "dummy_var",
          "value": "0",
        },
      ],
      "description": "",
      "foremanParameters": Array [
        Object {
          "description": "",
          "id": 1,
          "locked": false,
          "name": "CP",
          "type": "computeprofile",
          "value": "1",
        },
        Object {
          "description": "",
          "id": 2,
          "locked": true,
          "name": "LE",
          "type": "lifecycleenv",
          "value": "1",
        },
      ],
      "hostgroup": "1",
      "id": 1,
      "maxCount": "",
      "minCount": "2",
      "name": "web",
    },
    Object {
      "ansibleGroup": "dbservers",
      "ansibleParameters": Array [
        Object {
          "id": 0,
          "name": "mysqlservice",
          "value": "mysqld",
        },
        Object {
          "id": 1,
          "locked": true,
          "name": "mysql_port",
          "value": "3306",
        },
        Object {
          "id": 2,
          "name": "dbuser",
          "value": "webapp",
        },
        Object {
          "id": 3,
          "name": "dbname",
          "value": "ANSAP01",
        },
        Object {
          "id": 4,
          "name": "upassword",
          "value": "Bond@007",
        },
        Object {
          "id": 5,
          "name": "masterpassword",
          "value": "MySQL@007",
        },
      ],
      "backup": Object {
        "ansibleGroup": "dbservers",
        "ansibleParameters": Array [
          Object {
            "id": 0,
            "name": "mysqlservice",
            "value": "mysqld",
          },
          Object {
            "id": 1,
            "locked": true,
            "name": "mysql_port",
            "value": "3306",
          },
          Object {
            "id": 2,
            "name": "dbuser",
            "value": "webapp",
          },
          Object {
            "id": 3,
            "name": "dbname",
            "value": "ANSAP01",
          },
          Object {
            "id": 4,
            "name": "upassword",
            "value": "Bond@007",
          },
          Object {
            "id": 5,
            "name": "masterpassword",
            "value": "MySQL@007",
          },
        ],
        "description": "",
        "foremanParameters": Array [],
        "hostgroup": "1",
        "id": 2,
        "maxCount": "",
        "minCount": "1",
        "name": "db",
      },
      "description": "",
      "foremanParameters": Array [],
      "hostgroup": "1",
      "id": 2,
      "maxCount": "",
      "minCount": "1",
      "name": "db",
    },
  ],
  "columns": Array [
    Object {
      "cell": Object {
        "formatters": Array [
          null,
        ],
      },
      "header": Object {
        "formatters": Array [
          null,
        ],
        "label": "Service name",
        "props": Object {
          "index": 3,
          "style": Object {
            "width": "20%",
          },
        },
      },
      "property": "name",
    },
    Object {
      "cell": Object {
        "formatters": Array [
          null,
        ],
      },
      "header": Object {
        "formatters": Array [
          null,
        ],
        "label": "Hostgroup",
        "props": Object {
          "index": 2,
          "style": Object {
            "width": "20%",
          },
        },
      },
      "property": "hostgroup",
    },
  ],
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "name": false,
}
`;

exports[`ApplicationDefinitionImportReducer should return initial state 1`] = `
Object {
  "error": Object {
    "errorMsg": "",
    "status": "",
    "statusText": "",
  },
  "name": false,
}
`;
